File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/utils/MessageDigestUtils.java
package cn.tongdun.mobrisk.core.utils;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;

/**
 * @description: Message Digest utils
 * @author: wuzuchang
 * @date: 2022/12/6
 */
public class MessageDigestUtils {

    public static String hash(String algorithm, byte[] bytes) {
        try {
            MessageDigest md = MessageDigest.getInstance(algorithm);
            md.update(bytes);
            return StringUtils.byteToHexString(md.digest());
        } catch (Throwable ignored) {
        }
        return "";
    }

    public static String hash(String algorithm, String str) {
        try {
            MessageDigest md = MessageDigest.getInstance(algorithm);
            md.update(str.getBytes(StandardCharsets.UTF_8));
            return StringUtils.byteToHexString(md.digest());
        } catch (Throwable ignored) {
        }
        return "";
    }
}

File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/tools/MessageDigestUtils.kt
package cn.tongdun.mobrisk.core.tools

import java.nio.charset.StandardCharsets
import java.security.MessageDigest

/**
 * @description:
 * @author: wuzuchang
 * @date: 2023/5/12
 */

fun hash(algorithm: String, bytes: ByteArray): String {
    return executeSafe({
        val md = MessageDigest.getInstance(algorithm)
        md.update(bytes)
        StringUtils.byteToHexString(md.digest())
    }, "")
}

fun hash(algorithm: String, str: String): String {
    return executeSafe({
        val md = MessageDigest.getInstance(algorithm)
        md.update(str.toByteArray(StandardCharsets.UTF_8))
        StringUtils.byteToHexString(md.digest())
    }, "")
}